src/Security/LoginFormAuthenticator.php

```diff
@@ -333,7 +333,7 @@ public function start(Request $request, AuthenticationException $authException =
     {
         // Is this a JSON request?
         if (false !== strpos($request->getRequestFormat(), 'json')) {
-            throw new HttpException(Response::HTTP_UNAUTHORIZED, $authException->getMessage());
+            throw new HttpException(Response::HTTP_UNAUTHORIZED, 'Session expired! Please log in again!');
```

Comment:
This is a fine message. I think we should forward them to the login page, but maybe to serve this to the PO this way and see what they say.